tutorials/animation/index.md

```diff
@@ -0,0 +1,874 @@
+
+### AnimationController
+
+AnimationController is a special Animation object that generates a new
+value whenever the hardware is ready for a new frame. By default,
+an AnimationController linearly produces the numbers from 0.0 to 1.0
+during a given duration. For example, this code creates an Animation object,
+but does not start it running:
+
+<!-- skip -->
+{% prettify dart %}
+final AnimationController controller = new AnimationController(
+    duration: const Duration(milliseconds: 2000), vsync: this);
+{% endprettify %}
+
+AnimationController derives from Animation&lt;double&gt;, so it can be used
+wherever an Animation object is needed. However, the AnimationController
+has additional methods to control the animation. For example, you start
+an animation with the `.forward()` method. The generation of numbers is
+tied to the screen refresh, so typically 60 numbers are generated per
+second. After each number is generated, each Animation object calls the
+attached Listener objects. To create a custom display list for each
+child, see
+[RepaintBoundary](https://docs.flutter.io/flutter/widgets/RepaintBoundary-class.html).
+
+When creating an AnimationController, you pass it a `vsync` argument.
+This option keeps the current widget tree in display memory until
+Flutter's rendering engine completes its refresh cycle. To use your
+custom State object as the `vsync`, include the `TickerProviderStateMixin`
+when defining the custom State class.
+
+<aside class="alert alert-success" markdown="1">
+**Note**: In some cases, a position might exceed the AnimationController's
+0.0-1.0 range. For example, the `fling()` function allows you to provide
+velocity, force, and position (via the Force object).
+The position can be anything and so can be outside of the 0.0 to 1.0 range.
+
+A CurvedAnimation can also exceed the 0.0 to 1.0 range,
+even if the AnimationController doesn't. Depending on the curve selected,
+the output of the CurvedAnimation can have a wider range than the input.
+For example, elastic curves such as Curves.elasticIn will significantly
+overshoot or undershoot the default range.
+</aside>
+
+### CurvedAnimation
+
+A CurvedAnimation defines the animation's progress as a non-linear curve.
+
+<!-- skip -->
+{% prettify dart %}
+final CurvedAnimation curve =
+    new CurvedAnimation(parent: controller, curve: Curves.easeIn);
+{% endprettify %}
+
+<aside class="alert alert-success" markdown="1">
+**Note:**
+The [Curves](https://docs.flutter.io/flutter/animation/Curves-class.html)
+class defines many commonly used curves, or you can create your own.
+For example:
+
+<!-- skip -->
+{% prettify dart %}
+class ShakeCurve extends Curve {
+  @override
+  double transform(double t) {
+    return math.sin(t * math.PI * 2);
+  }
+}
+{% endprettify %}
+</aside>
+
+CurvedAnimation and AnimationController are both of type
+Animation&lt;double&gt;, so you can pass them interchangeably.
+Note that the CurvedAnimation wraps the object it’s modifying.
```

Comment:
nit: here and elsewhere, remove the words "note that" whenever they occur. They don't add anything.